src/main/cpp/RLBotInterface/BallPrediction/inc/mat.h

```diff
@@ -0,0 +1,350 @@
+#pragma once
+
+#include <math.h>
+#include <initializer_list>
+
+template < int N0, int N1 > 
+class mat {
+
+  float data[N0 * N1];
+
+  public:
+
+    mat(){} 
+
+    mat(float value){
+      for (int i = 0; i < N0 * N1; i++) {
+        data[i] = value;
+      }
+    } 
+
+    mat(std::initializer_list< std::initializer_list < float > > args){
+
+      int r = 0;
+      for (auto row : args) {
+        int c = 0;
+        for (auto val : row) {
+          operator()(r,c) = val;
+          c++;
+        }
+        r++;
+      }
+
+    }
+
+    float & operator[](const int i){
+      return data[i];
+    }
+
+    float operator[](const int i) const{
+      return data[i];
+    }
+
+    float & operator()(const int i0, const int i1){
+      return data[i0 + N0 * i1];
+    }
+
+    float operator()(const int i0, const int i1) const{
+      return data[i0 + N0 * i1];
+    }
+
+};
+
+template < int m, int n >
+inline mat < n, m > transpose(const mat < m, n > & A) {
```

Comment:
same here with brackets not matching everywhere

Reply:
yeah, most of my sources use this brace convention